mp/src/game/shared/neo/neo_gamerules.cpp

```diff
@@ -648,6 +666,88 @@ void CNEORules::Think(void)
 	bool bIsPause = m_nRoundStatus == NeoRoundStatus::Pause;
 	if (bIsIdleState && gpGlobals->curtime > m_flNeoNextRoundStartTime)
 	{
+		static int iStaticInitOnCmd = -1;
+		static int iStaticInitOnRandAllow = -1;
+
+		const int iGamemodeEnforce = neo_sv_gamemode_enforcement.GetInt();
+		const int iGamemodeRandAllow = neo_sv_gamemode_random_allow.GetInt();
+		// Update on what to select on first map load or server operator changes neo_sv_gamemode_enforcement
+		if (!m_bInitType || iGamemodeEnforce != iStaticInitOnCmd || iGamemodeRandAllow != iStaticInitOnRandAllow)
+		{
```

Comment:
Personal preference, but it might be worth abstracting this business of determining the mode into its own function, for readability, since we're currently nesting up to 6 scopes deep here with multiple if statements, a switch, for loops, static variable storage... it gets a bit difficult to keep track of control flow and state in one's head.